gulp-tasks/utils/package-runner.js

```diff
@@ -113,6 +113,7 @@ module.exports = (displayName, packageType, func, ...args) => {
   // We need to return a single no-op rather than an empty array, or else gulp
   // will throw 'One or more tasks should be combined using series or parallel'.
   if (packagePaths.length === 0) {
+    // TODO: should gulp handle this differently?
```

Comment:
There are scenarios where the flags result in us running gulp tasks that ultimately do nothing.

It's arguable what should happen here. The following snippet looks like developer error, but it's essentially what is happening here.

```javascript
gulp.series([])

// or

gulp.parallel([])
```

The other side effect is that we have to return a promise or use the callback since a synchronous function didn't work with gulp when I tried it.

Reply:
Yeah, that's tough. We want things to be usable without much ceremony (remember run-sequence, gulp-watch and gulp-plumber?) but we also don't want to introduce footguns like `gulp.series([])`.  I'm wondering if we can revisit the usage of this and see if there is a compromise between the 2 ends of the spectrum.

> The other side effect is that we have to return a promise or use the callback since a synchronous function didn't work with gulp when I tried it.

We did away with sync tasks because they generally were another footgun that people didn't realize.  All the benchmarks where people didn't return streams (and had sub-millisecond build times!) or had a runSequence but their clean task didn't finish before their build started were far too common.

Reply:
Maybe all tasks could just be marked as `async`?